src/FindAnApprenticeship/SFA.DAS.FindAnApprenticeship/Application/Queries/CreateAccount/SignIntoYourOldAccount/GetSignIntoYourOldAccountQueryHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.FindAnApprenticeship.InnerApi.LegacyApi.Requests;
+using SFA.DAS.FindAnApprenticeship.InnerApi.LegacyApi.Responses;
+using SFA.DAS.NServiceBus.Services;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.FindAnApprenticeship.Application.Queries.CreateAccount.SignIntoYourOldAccount;
+
+public class GetSignIntoYourOldAccountQueryHandler(
+    IFindApprenticeshipLegacyApiClient<FindApprenticeshipLegacyApiConfiguration> legacyApiClient,
+    ICacheStorageService cacheStorageService,
+    IDateTimeService dateTimeService)
+    : IRequestHandler<GetSignIntoYourOldAccountQuery, GetSignIntoYourOldAccountQueryResult>
+{
+    public async Task<GetSignIntoYourOldAccountQueryResult> Handle(GetSignIntoYourOldAccountQuery request, CancellationToken cancellationToken)
+    {
+        var cacheKey = GetCacheKey(request.CandidateId);
+        var cacheItem = await cacheStorageService.RetrieveFromCache<SignInAttemptHistory>(cacheKey);
+
+        if (cacheItem != null)
+        {
+            cacheItem.SignInAttempts.RemoveAll(x => x < dateTimeService.UtcNow.AddMinutes(-30));
+
+            if (cacheItem.SignInAttempts.Count >= 5)
+            {
+                return new GetSignIntoYourOldAccountQueryResult { IsValid = false };
+            }
+        }
+
+        var result =
```

Comment:
I think as well as doing this we also want to check the email hasnt already been migrated? unless this is done somewhere else